crates/matrix-sdk-crypto/src/machine.rs

```diff
@@ -1107,7 +1107,27 @@ impl OlmMachine {
             self.store.get_inbound_group_session(room_id, content.session_id()).await?
         {
             // TODO check the message index.
-            let (decrypted_event, _) = session.decrypt(event).await?;
+            let decrypt_result = session.decrypt(event).await;
+
+            if let Err(decrypt_error) = decrypt_result {
+                if let MegolmError::Decryption(ref vodo_error) = decrypt_error {
```

Comment:
The `ref vodo_error` is an oldish syntax, nowadays you would instead just borrow when destructuring `if let Foo::Bar(_) = &something`.

You can also destructure multiple layers of the enum in one `if let`:

```rust
if let Err(MegolmError::Decryption(_)) = &foo {
    ...
}
```